src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -45,29 +46,29 @@ public class BlockTree : IBlockTree
         private const int CacheSize = 64;
         private readonly LruCache<Keccak, Block> _blockCache = new LruCache<Keccak, Block>(CacheSize);
         private readonly LruCache<Keccak, BlockHeader> _headerCache = new LruCache<Keccak, BlockHeader>(CacheSize);
-        private readonly LruCache<long, ChainLevelInfo> _blockInfoCache = new LruCache<long, ChainLevelInfo>(CacheSize);
-
+        
         private const int BestKnownSearchLimit = 256_000_000;
         public const int DbLoadBatchSize = 4000;
 
         private long _currentDbLoadBatchEnd;
 
-        private ReaderWriterLockSlim _blockInfoLock = new ReaderWriterLockSlim();
-
         private object _batchInsertLock = new object();
 
         private readonly IDb _blockDb;
-
         private readonly IDb _headerDb;
+        private readonly IDb _blockInfoDb;
```

Comment:
? why not repository

Reply:
Repository is used only for ChainLevelInfo, BlockTree uses blockInfoDb also for additional things

Reply:
I am a bit worried about locking around this as it is now, to some extend managed in two different places...

Reply:
Renamed to ChainLevelInfoRepository to avoid confusion.